# Research Assignment 5 


Camellia Haghverdian

Factors Predicting Acceptance Rate

04/16/2024

In [1]:
// Setting up Data

insheet using "/Users/camellia/Documents/Stats 2/soc211_assignments/week10/loanrepay2013_highestresearch.csv", clear

(19 vars, 33 obs)


In [2]:
// Keeping only variables I need

keep acceptrate y_lo_inc_rpy sgrnt_low_~c 

In [3]:
// Turning dependent variable into dummy/dichotomous variable

tab acceptrate
tab acceptrate, nol

gen majority=acceptrate
replace majority=1 if acceptrate>=.50
replace majority=0 if acceptrate<=.50


 acceptrate |      Freq.     Percent        Cum.
------------+-----------------------------------
    .214635 |          1        3.03        3.03
    .228407 |          1        3.03        6.06
   .3206216 |          1        3.03        9.09
   .3395534 |          1        3.03       12.12
   .3987971 |          1        3.03       15.15
   .3991674 |          1        3.03       18.18
   .4189757 |          1        3.03       21.21
   .4269711 |          1        3.03       24.24
   .4531918 |          1        3.03       27.27
   .4872242 |          1        3.03       30.30
   .4924802 |          1        3.03       33.33
   .4995829 |          1        3.03       36.36
   .5001475 |          1        3.03       39.39
   .5190052 |          1        3.03       42.42
   .5257108 |          1        3.03       45.45
   .5543631 |          1        3.03       48.48
   .5946137 |          1        3.03       51.52
   .6090856 |          1        3.03       54.55
   .6095234 |      

In [4]:
tab majority


   majority |      Freq.     Percent        Cum.
------------+-----------------------------------
          0 |         12       36.36       36.36
          1 |         21       63.64      100.00
------------+-----------------------------------
      Total |         33      100.00


In [7]:
// Labeling dependent variable

label variable majority "Less Prestigious Schools"
label define majority 0 "Accepts less than 50% of applicants" 1 "Accepts more than 50% of applicants"
label values majority majority

label majority already defined
r(110);


In [6]:
// Taking a quick look at my independent variables

tab y_lo_inc_rpy
tab sgrnt_low_~c 


Y_lo_inc_rp |
          y |      Freq.     Percent        Cum.
------------+-----------------------------------
    47.9304 |          1        3.03        3.03
    50.4668 |          1        3.03        6.06
     55.709 |          1        3.03        9.09
    56.1845 |          1        3.03       12.12
    57.0114 |          1        3.03       15.15
    57.2512 |          1        3.03       18.18
    57.5044 |          1        3.03       21.21
    58.3367 |          1        3.03       24.24
    60.4167 |          1        3.03       27.27
    60.8015 |          1        3.03       30.30
    61.0008 |          1        3.03       33.33
    62.6499 |          1        3.03       36.36
    63.0356 |          1        3.03       39.39
    63.6468 |          1        3.03       42.42
    64.5707 |          1        3.03       45.45
    66.5521 |          1        3.03       48.48
    67.2012 |          1        3.03       51.52
    68.2734 |          1        3.03       54.55
    6

### 1. Develop hypotheses that approach a nominal variable in your dataset as a dependent variable (DV). [1 pt]

* If your nominal variable has more than two categorical variables, such as political party affiliation categories of -- Democrat, Democratic Socialist, Green, Independent, or Republican -- develop hypotheses that treat being or not being in at least one of the categories as a dichotomous binomial dependent variable (DV).

* Even if you are not interested in any categorical outcomes for your research project, still develop an hypothesis for at least one nominal variable in your data as a dependent variable. Thinking about potential influences on that variable could help you see if there are intervening relationships or omitted variable biases in your primary outcome of interest.

Do the following when you write your hypothesis:

    A. Write a separate hypothesis for each dependent variable (DV) you want to analyze.
    
    B. For each dependent variable (DV) state the predicted direction of association between your dependent variable DV and independent variables (IVs) in your model. You can predict no relationship if you do not expect a relationship.
    
    C. Write a couple sentences about the theoretical reasons (prior knowledge / research) for each of your predictions

In [8]:
*A. I hypothesize that the provision of more state grant aids and higher rates of loan repayment result 
// in higher rates of acceptance in universities. My dependent variable is acceptance rate (acceptrate), and my independent variables are
// loan repayment (y_lo_inc_rpy) and state grant aid for low-income students (sgrnt_low_~c).

*B and C (my theory and intutition included :)). 
    *1) I predict that the relationship between state grant and acceptance rate is positive, meaning with every increase in the percentage
    // of state grants, there will be an increase in the acceptance rate of schools. However, whether or not those schools will be more prestigious
    // or less prestigious is hard to tell. My assumption is that any school receiving additional state grants would become more prestigious
    // as time goes by.

    *2) In addition, the relationship between loan repayment and acceptance rate is positive, meaning with every increase in the percentage
    // of loan repayment, there will be an increase in the acceptance rate of schools because more low-income students would be interested in
    // attending such universities. However, if more low-income students attend a particular university, there is a high chance that that university
    // might lose its prestige over time. 

### 2. Test your hypotheses [3 pts]

    A. Create a frequency table for your dependent variable (DV).

    B. Estimate an OLS model regressing your dichotomous dependent variable (DV) on your hypothesized independent variables (IVs) and store the results.
    
    C. Estimate an MLE logistic model regressing your dichotomous dependent variable (DV) on your hypothesized independent variables (IVs) and store the results.
    
    D. Use esttab to output the coefficients for your OLS and MLE logistic models with html formatting within your Jupyter Notebook.
    
    E. Use esttab to reoutput your coefficients as odd ratios.

In [7]:
*** PART A: Creating frequency table for dependent variable

tab majority


           Less Prestigious Schools |      Freq.     Percent        Cum.
------------------------------------+-----------------------------------
Accepts less than 50% of applicants |         12       36.36       36.36
Accepts more than 50% of applicants |         21       63.64      100.00
------------------------------------+-----------------------------------
                              Total |         33      100.00


In [8]:
// Clearing stored estimates

est clear

In [9]:
*** PART B: Estimating an OLS model and storing results
// Use "ro", short for robust, to get robust standard errors

eststo: reg majority y_lo_inc_rpy sgrnt_low_~c, ro


Linear regression                               Number of obs     =         33
                                                F(2, 30)          =       6.94
                                                Prob > F          =     0.0033
                                                R-squared         =     0.2634
                                                Root MSE          =     .43302

------------------------------------------------------------------------------
             |               Robust
    majority | Coefficient  std. err.      t    P>|t|     [95% conf. interval]
-------------+----------------------------------------------------------------
y_lo_inc_rpy |  -.0329642   .0106397    -3.10   0.004    -.0546934   -.0112349
sgrnt_low_~c |   8.80e-06   .0000174     0.51   0.617    -.0000268    .0000444
       _cons |   2.766701   .6065554     4.56   0.000      1.52795    4.005453
------------------------------------------------------------------------------
(est1 stored)


In [11]:
*** PART C: Estimating an MLE logistic model and storing results
// Use "ro" again to get robust standard errors

eststo: logit majority y_lo_inc_rpy sgrnt_low_~c, ro
eststo: margins, dydx(*) post atmeans


Iteration 0:  Log pseudolikelihood = -21.630899  
Iteration 1:  Log pseudolikelihood =  -16.78412  
Iteration 2:  Log pseudolikelihood = -16.546006  
Iteration 3:  Log pseudolikelihood = -16.545214  
Iteration 4:  Log pseudolikelihood = -16.545214  

Logistic regression                                     Number of obs =     33
                                                        Wald chi2(2)  =   8.42
                                                        Prob > chi2   = 0.0149
Log pseudolikelihood = -16.545214                       Pseudo R2     = 0.2351

------------------------------------------------------------------------------
             |               Robust
    majority | Coefficient  std. err.      z    P>|z|     [95% conf. interval]
-------------+----------------------------------------------------------------
y_lo_inc_rpy |  -.1930458   .0759509    -2.54   0.011    -.3419069   -.0441848
sgrnt_low_~c |   .0000452   .0000789     0.57   0.566    -.0001093    .0001998


In [16]:
*** PART D: Outputting Table Results

esttab ///
, cells(b(star fmt(2)) se(fmt(2) par)) stardetach  ///
	legend starlevels(^ .1 * .05 ** .01 *** .001) ///
mlabels("OLS" "MLE / logistic" "MLE marginal effects") title("Unexpontiated Coefficients") ///
collabels(none)


Unexpontiated Coefficients
---------------------------------------------------------------
                      (1)              (2)              (3)    
                      OLS     MLE / logi~c     MLE margin~s    
---------------------------------------------------------------
main                                                           
y_lo_inc_rpy        -0.03 **         -0.19 *          -0.04 *  
                   (0.01)           (0.08)           (0.02)    
sgrnt_low_~c         0.00             0.00             0.00    
                   (0.00)           (0.00)           (0.00)    
_cons                2.77 ***        13.36 **                  
                   (0.61)           (4.82)                     
---------------------------------------------------------------
N                      33               33               33    
---------------------------------------------------------------
^ p<.1, * p<.05, ** p<.01, *** p<.001


In [17]:
*** PART E: Outputting Table Results with Odds Ratio
// Do this by using "eform"

esttab ///
, cells(b(star fmt(2)) se(fmt(2) par)) stardetach  ///
	legend starlevels(^ .1 * .05 ** .01 *** .001) ///
mlabels("OLS" "MLE / logistic" "MLE marginal effects") title("Odds Ratio") ///
collabels(none) eform


Odds Ratio
---------------------------------------------------------------
                      (1)              (2)              (3)    
                      OLS     MLE / logi~c     MLE margin~s    
---------------------------------------------------------------
main                                                           
y_lo_inc_rpy         0.97 **          0.82 *           0.96 *  
                   (0.01)           (0.06)           (0.02)    
sgrnt_low_~c         1.00             1.00             1.00    
                   (0.00)           (0.00)           (0.00)    
---------------------------------------------------------------
N                      33               33               33    
---------------------------------------------------------------
Exponentiated coefficients
^ p<.1, * p<.05, ** p<.01, *** p<.001


### 3. Interpret your results [1 pt]

Write a couple sentences answering each of the following questions:

    A. How would you interpret the relationship between your dependent variable(s) (DVs) and independent variables (IVs) based on the odds ratios for you MLE logistic model?

    B. Do the overall estimates support or contradict your hypotheses? Why?
    
    C. Is there anything suprising in your model estimates? Why?
    
    D. How do the OLS coefficients and their standard errors compare to the MLE estimates? Are the coefficients and standard errors larger or smaller?

QUESTION ????
*A: Looking at the odds ratio table and the marginal effects column with regards to the MLE model for the loan repayment, I can see that the results are statistically significant. The odds ratio for loan repayment is 0.96 (co-standard error 0.02), the percentage of which would be -0.04%. However, the odds ratio for state grant aids is 1.00 (co-standard error 0.00), 0%, which is an indicator that state grant aid (ind-v) has either no effect or a very small/negligible effect on acceptance rate (d-v).

*B: My hypothesis: I hypothesize that the provision of more state grant aids and higher rates of loan repayment result 
// in higher rates of acceptance in universities. My dependent variable is acceptance rate (acceptrate), and my independent variables are
// loan repayment (y_lo_inc_rpy) and state grant aid for low-income students (sgrnt_low_~c).

According to the results, with every increase in the loan repayment percentage, acceptance rate decreases, and therefore, (I think?) my hypothesis is rejected.

*C: I assumed there was going to be an increase in acceptance rates with the increase of loan repayment and state grants aids, but actually the odds ratio table shows that there is a decrease in acceptance rates in relation with loan repayment and the results on state grants showed negligible significance.

*D: I can see that the OLS model has a lower standard error of 0.01 compared to the MLE estimates that has the standard error of 0.02. Both standard errors for the OLS and MLE models for state grants are 0.00.